app/controllers/api/v1/admin/products_controller.rb

```diff
@@ -0,0 +1,70 @@
+class Admin::ProductsController < Api::V1::ApplicationController
+  # before_action :logged_in_user, only: [:edit, :update]
+  # before_action :admin_user, only: [:edit, :update]
+  before_action :authorize_admin, only: [:edit, :update]
+
+  def show
+    @product = Product.find(params[:id])
+    render json: @product
+  end
+
+  def new
+    @product = Product.new
+    render json: @product
+  end
+
+  def create
+    @product = Product.new(product_params)
+    if @product.save
+      flash[:success] = 'Product created'
+      redirect_to admin_product_path(@product)
```

Comment:
isn't this an API? this is not API specific code the flash, the redirect, the render new.